src/app/services/product.service.ts

```diff
@@ -12,89 +13,72 @@ export class ProductService {
   private productCountKey = 'productCount';
   private cartProductsKey = 'cartProducts';
 
-  selectedProduct: ProductData | null = null;
-  cartProducts: cartProductData[] = [];
-  productCount: number = 0;
-  activeButton: string = '';
+  public selectedProduct: ProductData | null = null;
+  public cartProducts: CartProductData[] = [];
+  public productCount = 0;
 
-  private apiurl = environment.apiUrl
+  private apiurl = environment.apiUrl;
 
-  constructor(private http: HttpClient) {
+  constructor(private http: HttpClient, private localStorageService: LocalStorageService) {
     this.loadFromLocalStorage();
   }
 
   private loadFromLocalStorage(): void {
-    const storedProduct = localStorage.getItem(this.selectedProductKey);
-    if (storedProduct) {
-      this.selectedProduct = JSON.parse(storedProduct);
-    }
-
-    const storedCartProducts = localStorage.getItem(this.cartProductsKey);
-    if (storedCartProducts) {
-      this.cartProducts = JSON.parse(storedCartProducts);
-    }
-
-    const storedProductCount = localStorage.getItem(this.productCountKey);
-    if (storedProductCount) {
-      this.productCount = JSON.parse(storedProductCount);
-    }
+    this.selectedProduct = this.localStorageService.getItem<ProductData>(this.selectedProductKey);
+    this.cartProducts = this.localStorageService.getItem<CartProductData[]>(this.cartProductsKey) || [];
+    this.productCount = this.localStorageService.getItem<number>(this.productCountKey) || 0;
   }
 
   private updateLocalStorage(): void {
-    localStorage.setItem(
-      this.selectedProductKey,
-      JSON.stringify(this.selectedProduct)
-    );
-    localStorage.setItem(
-      this.cartProductsKey,
-      JSON.stringify(this.cartProducts)
-    );
-    localStorage.setItem(
-      this.productCountKey,
-      JSON.stringify(this.productCount)
+    this.localStorageService.setItem(this.selectedProductKey, this.selectedProduct);
+    this.localStorageService.setItem(this.cartProductsKey, this.cartProducts);
+    this.localStorageService.setItem(this.productCountKey, this.productCount);
+  }
+
+  public getProducts(): Observable<ProductData[]> {
+    return this.http.get<ProductData[]>(this.apiurl).pipe(
+      map((response) => {
+        return response;
+      }),
+      catchError((error) => {
+        throw error;
+      })
     );
   }
 
-  getProducts(): Observable<ProductData[]> {
-    return this.http.get<ProductData[]>(this.apiurl);
+  public searchProducts(searchTerm: string): Observable<ProductData[]> {
```

Comment:
The endpoint does not provide search?

Reply:
Yeah

Reply:
I have removed the search logic form the product.services